source/lac/scalapack.cc

```diff
@@ -327,16 +327,29 @@ ScaLAPACKMatrix<NumberType>::copy_to(FullMatrix<NumberType> &matrix) const
   // we could move the following lines under the main loop above,
   // but they would be dependent on glob_i and glob_j, which
   // won't make it much prettier
-  if (property == LAPACKSupport::lower_triangular)
-    for (unsigned int i = 0; i < matrix.n(); ++i)
-      for (unsigned int j = i + 1; j < matrix.m(); ++j)
-        matrix(i, j) =
-          (state == LAPACKSupport::inverse_matrix ? matrix(j, i) : 0.);
-  else if (property == LAPACKSupport::upper_triangular)
-    for (unsigned int i = 0; i < matrix.n(); ++i)
-      for (unsigned int j = 0; j < i; ++j)
-        matrix(i, j) =
-          (state == LAPACKSupport::inverse_matrix ? matrix(j, i) : 0.);
+  if (state == LAPACKSupport::cholesky)
```

Comment:
why there these changes necessary? I also don't think it's right for the following reasons:

* if the source is in factorized state, you probably wold not try to copy it in the first place (not much use)
* if it's symmetric, then it's not triangular, so you should never get into the loop inside the `else if` part.

p.s. while there, please move matrix = 0.; outside of if (gird->mpi_process_is_active). It's a bug not to reset it on all cores.